diff --git a/lib/rules/compat.js b/lib/rules/compat.js
index 84ed5c3b1791e66eff331bc8184c2fd3960f9c3d..6910296bd5b5d04f8aeadf50b8b76f495813b061 100644
--- a/lib/rules/compat.js
+++ b/lib/rules/compat.js
@@ -137,13 +137,18 @@ exports.default = {
         const handleFailingRule = (node, eslintNode) => {
             if (isPolyfilled(context, node))
                 return;
-            errors.push({
-                node: eslintNode,
-                message: [
-                    generateErrorName(node),
+            const errorName = generateErrorName(node);
+            const needRequiresES6Flag = /fetch|IntersectionObserver|Promise|URL/.test(errorName);
+            const errorMessage = needRequiresES6Flag
+                ? `To use the ${errorName} method, first set "requiresES6" to true in the current gadget definition. Than, disable this warning by using "eslint-disable-next-line compat/compat"`
+                : [
+                    errorName,
                     "is not supported in",
                     node.getUnsupportedTargets(node, browserslistTargets).join(", "),
-                ].join(" "),
+                ].join(" ");
+            errors.push({
+                node: eslintNode,
+                message: errorMessage,
             });
         };
         const identifiers = new Set();
